### PR TITLE
avro-c++: Init at 1.7.5

### PR DESCRIPTION
**Pull request by rasendubi**
###### Things done
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/avro-c++/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl, cmake, boost155, python
```
Comment (joachifm) [inline pkgs/development/libraries/avro-c++/default.nix]: If you're only interested in the python interpreter, I believe the preferred method of getting at it is passing in `pythonPackages` and using `pythonPackages.python` in the build inputs.

File: pkgs/development/libraries/avro-c++/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}:
+
+stdenv.mkDerivation {
+  name = "avro-c++-1.7.5";
```
Comment (joachifm) [inline pkgs/development/libraries/avro-c++/default.nix]: To make updating a bit simpler, I recommend adding a `version` attribute and using it in the `name` & the `url` attrs.

File: pkgs/development/libraries/avro-c++/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  buildInputs = [
+    cmake
+    boost155
+    python
+  ];
+
+  enableParallelBuilding = true;
+
+  meta = {
```
Comment (joachifm) [inline pkgs/development/libraries/avro-c++/default.nix]: Please add yourself as the maintainer

**Comment by joachifm** [comment]
Apart from a few minor issues, this looks superficially okay

**Comment by rasendubi** [comment]
@joachifm, I've addressed all questions.

File: pkgs/development/libraries/avro-c++/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl, cmake, boost155, pythonPackages
+}:
+
+let version = "1.7.5"; in
```
Comment (joachifm) [inline pkgs/development/libraries/avro-c++/default.nix]: No need to update this, but the idiom is
```nix
stdenv.mkDerivation rec {
  name = "foo-${version}";
  version = "4.2";
}
```
Note the `rec` keyword allows you to refer to keys within the same attrset.

Comment (rasendubi) [inline pkgs/development/libraries/avro-c++/default.nix]: Yay, I know. I just looked through sample packages in nixpkgs manual and found next style is more common:

```nix
let version = "4.2"; in

stdenv.mkDerivation {
  name = "foo-${version}";
}
```

Comment (joachifm) [inline pkgs/development/libraries/avro-c++/default.nix]: `rec` is used way more, but it doesn't matter.

**Pull request merged**